test-five/frontend/cypress/e2e/spec.cy.ts

```diff
@@ -0,0 +1,8 @@
+
+describe('CRA', () => {
+  it('shows learn link', function () {
+    cy.visit('http://localhost:3000')
```

Comment:
is this going to work with CI/CD given it's hardcoded to localhost?

Reply:
It works within the CI/CD process as the plugin used in the workflow utilizes the secret environment [${{ secrets.FRONTEND_BASE_URL }}](https://github.com/pablomagro/show-us-what-you-got/blob/892c3f395fc63cb15c59cc1e4e640bfc04320e5d/.github/workflows/deploy.yml#L157), which is set to http://localhost:3000. This plugin installs, builds, and starts the server locally within the workflow.

Locally, it also functions when the local server is started. However, as a best practice, it should be referenced using an environment variable, as shown below:

```javascript
cy.visit(process.env.FRONTEND_BASE_URL)
```

Alternatively, you can achieve this by using a configuration library, such as config.ts, to read AWS SSM parameters for each environment. Here's an example:

```javascript
cy.visit(config.e2e.baseUrl)
```